ibm/service/scc/data_source_ibm_scc_posture_get_collector.go

```diff
@@ -0,0 +1,306 @@
+// Copyright IBM Corp. 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package scc
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/scc-go-sdk/v3/posturemanagementv2"
+)
+
+func DataSourceIBMSccPostureGetCollector() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMSccPostureGetCollectorRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
```

Comment:
Don't use "id" this a arguemnt reserved for Terraform core rename the attribute to collector_id
Same comments hold for other datasouces also